mypy/checker.py

```diff
@@ -366,6 +367,11 @@ def check_second_pass(self,
                             self.check_partial(node)
             return True
 
+    def prepare_types(self) -> None:
+        """Additional preparations before actual type checking."""
+        # We need to set `TypeType` fallback, since it is now unset:
+        TypeType.fallback = self.type_type()
```

Comment:
This is inconsistent with how fallbacks are set everywhere else. I don't think that we should use a global (or class) variable for this. Instead, we should look up the fallback when constructing an instance.

Reply:
Yes, exactly. This is a single variable, because `TypeType` has always a fallback to `builtins.type`. But, since `TypeType` instances are created quite commonly, I've decided to make it a class-variable, so we can skip addding `TypeType(inst, fallback=self.named_type('builtins.type'))` everywhere.

At some places in the code, we can't even do that: because `chk` instance is not available there.